src/nvim/os/pty_process_unix.c

```diff
@@ -73,6 +73,12 @@ int pty_process_spawn(PtyProcess *ptyproc)
     goto error;
   }
 
+  // Other jobs and providers shouldn't get an copy of this file descriptor.
+  if (os_set_cloexec(master) == -1) {
```

Comment:
@hardenedapple I'm confused. `os_set_cloexec` is checking `== -1` but based on your comment it sounds like it should check `< 0`.